src/job.rs

```diff
@@ -306,7 +306,32 @@ impl Job {
                     async_smtp::smtp::error::Error::Transient(_) => {
                         // We got a transient 4xx response from SMTP server.
                         // Give some time until the server-side error maybe goes away.
-                        Status::RetryLater
+                        
+                        // But let's first check if we didn't retry this job already
+                        // too often.
+                        let actually_permanent = self.tries >= JOB_RETRIES - 1;
```

Comment:
I don't really like this code here as i just copied the code from some lines above. So in case the logic/fix is approved to merge, overhauling this should be done.